File: StringToBinario/Python/Conversor.py
#       
#   Crea un programa se encargue de transformar una cadena de texto a binario 
#   sin utilizar funciones propias del lenguaje que lo hagan directamente.
#
#Creamos las variables que vamos a utilizar 
cadena = ["Hola mundo"]
vocales = ['a', 'e', 'i', 'o', 'u']
cadena_convertida = ""

# Recorremos las palabras en la lista
for palabra in cadena:
    # Inicializamos la cadena para cada palabra
    palabra_convertida = ""
    
    # Recorremos las letras de cada palabra
    for letra in palabra:
        # Verificamos si la letra es una vocal
        if letra.lower() in vocales:
            palabra_convertida += "1"
        else:
            palabra_convertida += "0"
    
    # Agregamos la palabra convertida a la cadena general
    cadena_convertida += palabra_convertida + " "

# Imprimimos la cadena convertida
print(cadena_convertida)

File: StringToBinario/C++/Conversor.cpp
/*
 * Crea un programa se encargue de transformar una cadena de texto a binario 
 * sin utilizar funciones propias del lenguaje que lo hagan directamente.
 */
#include<iostream>

using namespace std;

int main(int argc, char const *argv[])
{
    //Creamos las variables que vamos a necesitar para realizar el ejercicio
    string cadena[] = {"Hola mundo"};
    int numElementos = sizeof(cadena) / sizeof(cadena[0]);
    string cadenaConvertida = "";

    //Creamos un bucle que recorra la palabra
    for(int i = 0; i < numElementos; i++)
    {
        /* y otro que recorra las letras */
        for (int j = 0; j < cadena[i].length(); j++)
        {
            //Creamos un condicional que cada vez que identifique una vocal en la palabra, la convierta en 1
            if ((cadena[i][j] == 'a') || (cadena[i][j] == 'e') || (cadena[i][j] == 'i') || 
                (cadena[i][j] == 'o') || (cadena[i][j] == 'u')) 
            {
                cadenaConvertida += "1";     
            } 
            else 
            {
               //En caso contrario será un 0
                cadenaConvertida += "0";
            }
        }      
    }
    //Imprimimos el resultado
    cout << cadenaConvertida << endl;

    return 0;
}